### PR TITLE
R doc 1016

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+### How to run it?
+For Windows, the syntax to run will be:
+{CODE-BLOCK:powershell}
+Voron.Recovery.exe recover <Voron data-file directory> <Recovery directory>
+{CODE-BLOCK/}
+
+For Linux, the syntax to run will be:
+{CODE-BLOCK:powershell}
+dotnet Voron.Recovery.dll recover <Voron data-file directory> <Recovery directory>
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown]: Shouldn't this be `./Voron.Recovery recover ...` on linux?

File: Documentation/4.0/Raven.Documentation.Pages/users-issues/recovering-from-voron-errors.markdown [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  1. VoronUnrecoverableErrorException: `Index points to a non leaf page`  
+  2. VoronUnrecoverableErrorException: `Was unable to retrieve the correct node. Data corruption possible`  
+  3. VoronUnrecoverableErrorException: `Error syncing the data file. The last sync tx is...`  
+
+##Possible Causes:
+Such exceptions are caued by the corruption of a Voron data file.
+First, it is possible that the data file is corrupted due to critical bug.
+Also, data file corruption is likely to happen due to a hardware failure,if the file system doesn't support `fsync` functionality, or the storage hardware does not respect the `fsync` commands.
+
+{NOTE In Unix based OS it is a `fsync` command, in Windows OS, it is a matter of creating a file with a 'Write-Through' flag. /}
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/users-issues/recovering-from-voron-errors.markdown]: This is actually not quite true.
We have both `fsync` requirement and write-through unbuffered on Windows and `O_DIRECT` on Linux

File: Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿# Voron Recovery Tool
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown]: why is it in glossary section? we have `server/troubleshooting`

File: Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+{CODE-BLOCK:powershell}
+Voron.Recovery.exe recover <Voron data-file directory> <Recovery directory>
+{CODE-BLOCK/}
+
+For Linux, the syntax to run will be:
+{CODE-BLOCK:powershell}
+dotnet Voron.Recovery.dll recover <Voron data-file directory> <Recovery directory>
+{CODE-BLOCK/}
+
+The process may take some time to run, depending on how fast the storage is.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown]: and the size of the recovered database

File: Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown [hunk 5]
```diff
@@ -0,0 +1,40 @@
+﻿# Voron Recovery Tool
+
+### What is it?
+Voron recovery tool is designed to extract your data even on the worst corruption state imaginable.
+
+### How to run it?
+For Windows, the syntax to run will be:
+{CODE-BLOCK:powershell}
+Voron.Recovery.exe recover <Voron data-file directory> <Recovery directory>
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/glossary/voron-recovery-tool.markdown]: just to be sure: did you verify it's working. I had some issue running it this way:
https://github.com/ravendb/docs/pull/520#discussion_r162569017

**Comment by talweiss1982** [comment]
I'll do the modification, I also need to mention that the recovery tool may bring back deleted documents, we have not solved this issue.

**Comment by arekpalinski** [comment]
Yeah, that will be very important note

**Pull request merged**